### ISSUE TITLE
Linux compile error: 'QPair' was not declared in this scope

### ISSUE THREAD

**Issue by gzuk78** [opentoonz/opentoonz#3096]
Hi, I tried to follow the Linux compile instructions at https://github.com/opentoonz/opentoonz/blob/master/doc/how_to_build_linux.md

There were no errors or warnings regarding any dependencies. libmypaint and LibTIFF compiled just fine.

But the final OpenToonz build aborted with a compile error at tcolorutils.cpp.o. I think it may be because it's the master branch, which may be in development? Could someone please give me the "last stable" version to check out via git, so I can try that again? Or did I miss something else?

Many thanks!

The last commit in the git log is this:

```
commit 7a3a929f52e46eaa69c1cc4a72dadbfecd71dff6
Merge: 2d54639 b713bec
Author: Rodney <rodney.baker@gmail.com>
Date:   Wed Feb 5 08:57:15 2020 -0700

    Merge pull request #3095 from shun-iwasawa/fix_xdts_export_duration
    
    Fix XDTS Export Feature to Use Output Range for Duration
```

The compile error looks like this:

```
[ 14%] Building CXX object tnzbase/CMakeFiles/tnzbase.dir/__/common/tapptools/tcolorutils.cpp.o
In file included from /home/gzuk/mysoftware/opentoonz/toonz/sources/common/tapptools/tcolorutils.cpp:4:0:
/home/gzuk/mysoftware/opentoonz/toonz/sources/include/tcolorutils.h:32:40: error: 'QPair' was not declared in this scope
 DVAPI void buildColorChipPalette(QList<QPair<TPixel32, TPoint>> &palette,
                                        ^
/home/gzuk/mysoftware/opentoonz/toonz/sources/include/tcolorutils.h:32:56: error: wrong number of template arguments (2, should be 1)
 DVAPI void buildColorChipPalette(QList<QPair<TPixel32, TPoint>> &palette,
                                                        ^
```

**Comment by RodneyBaker** [comment]
Hi @gzuk78 
If you go here:  https://github.com/opentoonz/opentoonz
Over to the right you'll see a green button for cloning or downloading the current master.

Sorry to say, I don't have any insights into your compiling error.  
Perhaps someone that builds on Linux will be able to assist.

**Comment by Lovewyrm** [comment]
current master (aka OPs a3a929f52e46eaa69c1cc4a72dadbfecd71dff6 ), just built flawlessly for me.
Slackware linux 64 bit (multilib) on the 'current' snapshot (Slackware has stable releases often taking years, with the 'current' one being the 'rolling' state between those).

My qt5 version is 5.9.9

**Comment by gzuk78** [comment]
Thanks for the info, Lovewyrm.

I checked out opentoonz tag v1.4.0, but the error is still there. So it really must to be something in my setup. QPair seems to be a Qt class, so maybe it's the Qt version? My qt5 version is 5.5.1, but my Mint installation is so old that the package repository doesn't offer any newer version out of the box.

I am using the OTX AppImage binary now. That seems to work, though it crashes a lot.

**Comment by manongjohn** [comment]
I think you need at least QT 5.9 to compile OT which is likely why you are having issues. You can manually download 5.9 from the QT site and see if that fixes your compile issue.

Not sure how well OTX works on other Linux variants. May work better on some vs others. OTX is built on Ubuntu 16.04 so if your Mint is older than the equivalent of that, it may be more unstable.

**Comment by gzuk78** [comment]
Yes, it seems to be the Qt version. I ran the same binary on a newer Mint installation, and it worked flawlessly. I then tried to install Qt 5.11 on my old Mint, but it gets installed under a different ID and path, `/opt/qt511`, and doesn't replace the old one. I haven't yet figured out how to pass this new path to the OSX binary or the cmake compile setup. The Qt docs say to create a file `~/.config/qtchooser`, but that's only picked up by Qt applications such as qmake and creator.

**Comment by manongjohn** [comment]
Try this:

`$ QT_PATH=/opt/qt511/lib cmake ../sources`

Might need to play with the /opt/qt511/lib path but I believe that shouls override the default QT_PATH it is trying to configure the build scripts with.

**Comment by RodneyBaker** [comment]
Closing this as likely resolved although unconfirmed.

Recommend opening a new report with examples, video, project files in order to lock the issue down and target it for resolution.

Recommend referencing this issue in any newly opened report.
